src/main/java/io/split/android/client/SplitFactoryHelper.java

```diff
@@ -70,15 +70,24 @@ SplitStorageContainer buildStorageContainer(Context context, Key key, String dat
                 StorageFactory.getPersistenImpressionsStorage(splitRoomDatabase));
     }
 
+    String buildSplitsFilterQueryString(SplitClientConfig config) {
+        SyncConfig syncConfig = config.syncConfig();
+        if(syncConfig != null) {
+            return new FilterBuilder().addFilters(syncConfig.getFilters()).build();
```

Comment:
as a suggestion, why don't we use FilterBuilder inside SyncConfig, and change this method as follows:

```
String buildSplitsFilterQueryString(SplitClientConfig config) {
        SyncConfig syncConfig = config.syncConfig();
        if(syncConfig != null) {
                  return syncConfig.getSplitFiltersQuery();
        }
        return null;
}
```

This way you decouple your code, because SplitFactoryHelper only depends on SyncConfig (doesn't need to know FilterBuilder), and also you can better test how query strings are built by unit testing SyncConfig: it receives as input a list of user defined filters and it returns the query string.

Reply:
already discussed in zoom